### PR TITLE
Proposals and changes

### PR DESCRIPTION
**Pull request by tachithanhdanh**
The comments I wrote are explanatory enough, aren't they?

### CODE REVIEW

File: Template/template.cpp [hunk 1]
```diff
@@ -2,40 +2,51 @@
...
-#define FOR(i,a,b) for (int i=int(a); i<=int(b); i++)
-#define ROF(i,a,b) for (int i=int(a); i>=int(b); i--)
-#define each(i,a) for(auto &i:a)
+// use postfix increment/decrement operator for speed and better optimization
+// http://www.cs.technion.ac.il/users/yechiel/CS/FAQs/c++-faq/increment-pre-post-speed.html
+#define FOR(i,a,b) for (int i = (a); i <= (b); ++i) // Less than or equal?, should be less than
+#define ROF(i,a,b) for (int i = (a); i >= (b); --i) // (a)-1?
+// No need to cast to int lol
+#define each(i,a) for(auto& i : a) // C++ 11 and after
+// You shoud add some spaces to enhance readability both for you and for others
```
Comment (hoangdaoviet) [inline Template/template.cpp:30]: I shouldn't make a template for for loop. Sometimes it's difficult to control the variables.

Comment (hoangdaoviet) [reply Template/template.cpp:30]: Like the "less than or equal" as you said.

Comment (hoangdaoviet) [reply Template/template.cpp:30]: Btw, the postfix is a good thing. :D

Comment (tachithanhdanh) [reply Template/template.cpp:30]: > Btw, the postfix is a good thing. :D

as always

**Pull request merged**